addons/web_tour/static/src/js/running_tour_action_helper.js

```diff
@@ -148,12 +148,43 @@ var RunningTourActionHelper = core.Class.extend({
             $to = get_jquery_element_from_selector(to);
         }
 
-        let toCenter = calculateCenter();
+        let toCenter = this._calculateCenter($to, to);
         values.$element.trigger($.Event("mousemove", {which: 1, pageX: toCenter.left, pageY: toCenter.top}));
         // Recalculate the center as the mousemove might have made the element bigger.
-        toCenter = calculateCenter();
+        toCenter = this._calculateCenter($to, to);
         values.$element.trigger($.Event("mouseup", {which: 1, pageX: toCenter.left, pageY: toCenter.top}));
-     },
+    },
+    _drag_move_and_drop: function (values, to) {
+        let initialMoveOffset = [1, 0];
+        let fromSelector;
+        const elementCenter = values.$element.offset();
+        elementCenter.left += values.$element.outerWidth() / 2;
+        elementCenter.top += values.$element.outerHeight() / 2;
+        // Check if initial move offset is defined: '[deltaX,deltaY]@from => actualTo'.
+        const parts = /^\[(.+),(.+)\]@(.+) => (.+)/.exec(to);            
+        if (parts) {
+            initialMoveOffset[0] = parseInt(parts[1]);
+            initialMoveOffset[1] = parseInt(parts[2]);
+            fromSelector = parts[3];
+            to = parts[4];
+        }
+
+        values.$element.trigger($.Event("mouseenter"));
+        values.$element.trigger($.Event("mousedown", {which: 1, pageX: elementCenter.left, pageY: elementCenter.top}));
+        let relative = elementCenter;
```

Comment:
Seems useless no?
Reassign 2 lines bellow
```js
relative = $from.offset();
```

Reply:
Reorganized it all, I think it is way more readable now.